### ISSUE TITLE
tfp.distributions.Categorical(logits=[np.NAN, np.NAN, np.NAN]).sample() returns 3

### ISSUE THREAD

**Issue by LukeWood** [tensorflow/probability#1571]
tfp.distributions.Categorical(logits=[np.NAN, np.NAN, np.NAN]).sample() returns 3 instead of throwing an error

**Comment by csuter** [comment]
I expect you'll get an error if you construct the instance with
validate_args=True. Most validations are off by default in tfp because they
slow things down a lot. The idea is that it is fast by default, but has (a
ton of) correctness checks when enabled, for debugging purposes.
Categorical basically just does some shape munging and them defers to
tf.random.categorical; this behavior is probably bubbling up from there.

**Comment by LukeWood** [comment]
Thanks for letting me know.  I really appreciate it.

**Comment by LukeWood** [comment]
Is it too late as a library to make `validate_args=True` the default?  I implemented REINFORCE (aka vanilla policy gradient) in both PyTorch and TensorFlow this weekend, and this turned out to be a large debugging issue.  

I'm sure I lack some context and background here, but my intuition is that UX is more important than defaulting to the performant version here.

All in all, the experience was better in TensorFlow than in Torch, though I am incredibly biased as a Keras team member.  cheers

**Comment by csuter** [comment]
I think it's too late. I think the original thinking was that we could
optimize for newcomer ux (a (hopefully) transient state) or expert ux (a
(hopefully) steady state), and we opted for the latter. It was a conscious
trade-off, in the hopes that we could fill the gap with tutorials and
documentation. I think we could always do better on the education front,
but this particular issue actually comes up surprisingly rarely (though
certainly not never), all things considered!
